modules/mapbox/src/mapbox-overlay.ts

```diff
@@ -5,28 +5,50 @@ import type {Map, IControl, MapMouseEvent} from 'mapbox-gl';
 import type {MjolnirGestureEvent, MjolnirPointerEvent} from 'mjolnir.js';
 import type {DeckProps} from '@deck.gl/core';
 
+import {resolveLayers} from './resolve-layers';
+
+export type MapboxOverlayProps = Omit<
+  DeckProps,
+  | 'width'
+  | 'height'
+  | 'gl'
+  | 'parent'
+  | 'canvas'
+  | '_customRender'
+  | 'viewState'
+  | 'initialViewState'
+  | 'controller'
+>;
+
 /**
  * Implements Mapbox [IControl](https://docs.mapbox.com/mapbox-gl-js/api/markers/#icontrol) interface
  * Renders deck.gl layers over the base map and automatically synchronizes with the map's camera
  */
 export default class MapboxOverlay implements IControl {
-  private _props: DeckProps;
+  private _props: MapboxOverlayProps;
   private _deck?: Deck;
   private _map?: Map;
   private _container?: HTMLDivElement;
+  private _interleaved: boolean;
 
-  constructor(props: DeckProps) {
-    this._props = {...props};
+  constructor(
+    props: MapboxOverlayProps & {
+      interleaved?: boolean;
+    }
+  ) {
+    const {interleaved = false, ...otherProps} = props;
```

Comment:
The default for `GoogleMapsOverlay` is `interleaved=true`. Would it not be better to be consistent?

Reply:
Is google's WebGL2? I thought one reason we didn't do it by default was mapbox is WebGL, which for some cases could be limiting (though I don't notice which cases, so this could be an old assumption).

Are we positioning MapboxOverlay as the de-facto way of using mapboxgl-js (or react-map-gl) with deck now? If yes, I'm hesitant to default interleaved rendering for the reason stated above. If no, then default interleaved would make sense to me since the user is opting in to a tight mapbox ecosystem integration.

Reply:
Mapbox's interleaving implementation is less reliable than Google Map's. mapbox-gl's WebGLContext is WebGL1 and there are known issues of z-fighting between native layers and deck layers. Since we are promoting this solution I think it's safer to have the option off for now. We can revisit the decision in the next release, after getting some feedback from users.